crates/libs/nuts/src/nut12/mod.rs

```diff
@@ -0,0 +1,347 @@
+//! NUT-12: Offline ecash signature validation (DLEQ proofs)
+// https://cashubtc.github.io/nuts/12/
+
+use bitcoin::hashes::{Hash, HashEngine, sha256::Hash as Sha256Hash};
+use bitcoin::secp256k1::{PublicKey as SecpPublicKey, Scalar, SecretKey as SecpSecretKey};
+
+use crate::SECP256K1;
+use crate::dhke::{self, Error};
```

Comment:
create an `Error` type in this file
and import it in dhke behind feature flag as nut12::Error